File: SimboDev24/sample/javascript/Tennis.js
const hitsToScore = new Map();

hitsToScore.set(0, "Love");
hitsToScore.set(1, "Fifteen");
hitsToScore.set(2, "Thirty");
hitsToScore.set(3, "Forty");

function Tennis(aName, bName) {

    this.aName = aName;
    this.bName = bName;

    this.aHits = 0;
    this.bHits = 0;

    this.a = function () {
        this.aHits++;
    };

    this.b = function () {
        this.bHits++;
    };

    this.score = function () {

        if (this.aHits === this.bHits) {
            return this.aHits >= 3
                ? 'Deuce'
                : hitsToScore.get(this.aHits) + ' All';
        } else {

            if (this.aHits > 3) {
                return this.aName + ' Adv.';
            } else if (this.bHits > 3) {
                return this.bName + ' Adv.';

            } else {
                return hitsToScore.get(this.aHits) + ' ' + hitsToScore.get(this.bHits);
            }
        }


    }


}


module.exports = Tennis;

File: SimboDev24/sample/java/src/main/java/Tennis.java
import java.util.HashMap;
import java.util.Map;

public class Tennis {
    private static Map<Integer, String> scoreLookup = new HashMap<>();

    static {
        scoreLookup.put(0, "Love");
        scoreLookup.put(1, "Fifteen");
        scoreLookup.put(2, "Thirty");
        scoreLookup.put(3, "Forty");
    }

    private int aScore = 0;
    private int bScore = 0;

    public String score() {
        if (aScore != bScore) {
            return scoreLookup.get(aScore) + " " + scoreLookup.get(bScore);
        }
        if (aScore > 0) {
            return "Fifteen All";
        }
        return "Love All";
    }

    public void firstPlayerHit() {
        this.aScore++;
    }

    public void bHit() {
        this.bScore++;
    }
}
